cli/flox-rust-sdk/src/models/manifest.rs

```diff
@@ -500,12 +553,101 @@ pub struct ManifestInstall(
     BTreeMap<String, ManifestPackageDescriptor>,
 );
 
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
+#[cfg_attr(test, derive(proptest_derive::Arbitrary))]
+// todo: this can make the error messages less clear and might call for a custom (de)serialize impl
```

Comment:
blocking: do we need to address this before merging? What do our errors look like with this change?

Reply:
```
[install]
hello.unknown = ""
```

currently produces (± some traces that we can get patched):

```
unknown field `unknown`, expected one of `pkg-path`, `pkg-group`, `priority`, `version`, `systems`
in `install.hello`
```


with this change (because we have no key to distinguish variants) we get:

```
data did not match any variant of untagged enum ManifestPackageDescriptor
in `install.hello`
```

With the `serde(expecting = ...)` attribute we can get something somewhat better, e.g.:

```
❌ ERROR: Failed to parse manifest:

Expected either a catalog or a flake descriptor
in `install.hello`
```

and `serde-untagged` allows to format that a bit nicer outside of a derive macro string.

`serde_path_to_error` is sometimes suggested as well but it deosnt seem it would help in our case as the error is still somewhat the same.


In fact the issue with untagged enums has been raised and PRed agains at least twice in serde upstream: https://github.com/serde-rs/serde/pull/1544 and https://github.com/serde-rs/serde/pull/2376 but both PRs were rejected unfortunately

Reply:
the most user friendly we can do is providing an `expecting` attribute. 
a backwards incompatible change, would be requiring a type attribute but while that gives more detailed errors its also annoying to type for every install.

Lastly, we could separate flake installs into a separate table similar to what @mkenigs suggested for the lockfile